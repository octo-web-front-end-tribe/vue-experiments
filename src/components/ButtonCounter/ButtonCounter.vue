<template>
  <button @click="incrementCounter" class="test">{{ counter }}</button>
</template>

<script>
  export default {
    data () {
      return {
        counter: 0
      }
    },
    methods: {
      incrementCounter: function () {
        this.counter += 1
        this.$emit('increment')
      }
    }
  }
</script>

<style scoped>
  .test {
    font-size: 30px;
    color: red;
  }
</style>
