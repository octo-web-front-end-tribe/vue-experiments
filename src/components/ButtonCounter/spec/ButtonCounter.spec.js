import {shallow} from 'vue-test-utils';
import ButtonCounter from '../ButtonCounter';

describe('ButtonCounter.vue', () => {
  it('should match snapshot', () => {
    // when
    const wrapper = shallow(ButtonCounter);

    // then
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should initialize counter to 0', () => {
    // when
    const wrapper = shallow(ButtonCounter);

    // then
    let button = wrapper.find('.test');
    expect(button.text()).toEqual('0');
  });

  it('should increment counter on click', () => {
    // given
    const wrapper = shallow(ButtonCounter);
    let button = wrapper.find('.test');

    // when
    button.trigger('click');

    // then
    expect(button.text()).toEqual('1');
  });

  it('should emit an event increment on click', () => {
    // given
    const wrapper = shallow(ButtonCounter);
    let button = wrapper.find('.test');
    let spy = jest.fn();
    wrapper.vm.$on('increment', spy);

    // when
    button.trigger('click');

    // then
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
