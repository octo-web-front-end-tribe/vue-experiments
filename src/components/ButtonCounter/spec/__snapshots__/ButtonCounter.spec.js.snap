// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCounter.vue should match snapshot 1`] = `
<button
  class="test"
>
  0
</button>
`;
